app/services/layout.ts

```diff
@@ -15,63 +22,135 @@ export enum ELayoutElement {
   Sources = 'Sources',
 }
 
+export type LayoutSlot = '1' | '2' | '3' | '4' | '5' | '6';
+
 interface ILayoutServiceState {
   currentLayout: ELayout;
-  slottedWidgets: { [key in ELayoutElement]?: '1' | '2' | '3' | '4' | '5' | '6' };
-  resizes: { bar1: number; bar2: number };
+  slottedElements: { [value in ELayoutElement]?: LayoutSlot };
+  resizes: { bar1: number; bar2?: number };
 }
 
 const RESIZE_DEFAULTS = {
-  [ELayout.Default]: {
-    bar1: 156,
-    bar2: 240,
-  },
-  [ELayout.TwoPane]: {
-    bar1: 650,
-    bar2: 300,
-  },
+  [ELayout.Default]: { bar1: 156, bar2: 240 },
+  [ELayout.TwoPane]: { bar1: 650, bar2: 300 },
+  [ELayout.Classic]: { bar1: 450 },
+  [ELayout.FourByFour]: { bar1: 170, bar2: 170 },
+  [ELayout.Triplets]: { bar1: 650, bar2: 300 },
+  [ELayout.OnePane]: { bar1: 500 },
+};
+
+const ELEMENT_MINS = {
+  [ELayoutElement.Display]: { x: 350, y: 250 },
+  [ELayoutElement.LegacyEvents]: { x: 430, y: 150 },
+  [ELayoutElement.Mixer]: { x: 230, y: 150 },
+  [ELayoutElement.Minifeed]: { x: 330, y: 150 },
+  [ELayoutElement.Sources]: { x: 220, y: 150 },
+  [ELayoutElement.Scenes]: { x: 200, y: 150 },
```

Comment:
I think a lot of this stuff should really be pushed into the element classes themselves.  If they all subclass a common abstract class or implement a common interface, you can require all elements to define these values in their implementation.  This is cleaner and is better OO design, because you are clearly defining the requirements/shape of what a element implementation looks like.  It's much easier for a developer down the road to add a new element.  I think this comment also applies to layouts.  Things like the name of the layout should be an attribute on the layout itself rather than in a map object.